### ISSUE TITLE
7.14.0: broken `setuptools`<>`sphinx` integration and sphinx warnings

### ISSUE THREAD

**Issue by kloczek** [elastic/elasticsearch-py#1704]
Looks like `setuptools`<>`sphinx integration is broken:
```console
+ /usr/bin/python3 setup.py build_sphinx -b man --build-dir build/sphinx
running build_sphinx
Running Sphinx v4.1.2

Configuration error:
config directory doesn't contain a conf.py file (/home/tkloczko/rpmbuild/BUILD/elasticsearch-py-7.14.0/docs)
```
Patch for that:
```patch
--- a/setup.cfg~        2021-09-01 12:52:39.000000000 +0100
+++ b/setup.cfg 2021-09-01 12:53:23.559824969 +0100
@@ -1,6 +1,5 @@
 [build_sphinx]
-source-dir = docs/
-build-dir = docs/_build
+source-dir = docs/sphinx
 all_files = 1

 [bdist_wheel]

```
Feel free to commit that or let me know if you want to have that as PR.
After fixing that I found that current sphinx 4.1.2 shows many warnings.
```console
+ /usr/bin/python3 setup.py build_sphinx -b man --build-dir build/sphinx
running build_sphinx
Running Sphinx v4.1.2
making output directory... done
building [mo]: all of 0 po files
building [man]: all source files
updating environment: [new config] 7 added, 0 changed, 0 removed
reading sources... [100%] transports
/home/tkloczko/rpmbuild/BUILD/elasticsearch-py-7.14.0/elasticsearch/client/__init__.py:docstring of elasticsearch.client.Elasticsearch.index:15: WARNING: Inline interpreted text or phrase reference start-string without end-string.
/home/tkloczko/rpmbuild/BUILD/elasticsearch-py-7.14.0/docs/sphinx/async.rst:4: WARNING: duplicate object description of elasticsearch, other instance in api, use :noindex: for one of them
/home/tkloczko/rpmbuild/BUILD/elasticsearch-py-7.14.0/docs/sphinx/async.rst:208: WARNING: duplicate object description of elasticsearch, other instance in async, use :noindex: for one of them
/home/tkloczko/rpmbuild/BUILD/elasticsearch-py-7.14.0/elasticsearch/_async/client/__init__.py:docstring of elasticsearch._async.client.AsyncElasticsearch.index:15: WARNING: Inline interpreted text or phrase reference start-string without end-string.
/home/tkloczko/rpmbuild/BUILD/elasticsearch-py-7.14.0/docs/sphinx/connection.rst:22: WARNING: duplicate object description of elasticsearch, other instance in async, use :noindex: for one of them
/home/tkloczko/rpmbuild/BUILD/elasticsearch-py-7.14.0/docs/sphinx/exceptions.rst:6: WARNING: duplicate object description of elasticsearch, other instance in connection, use :noindex: for one of them
/home/tkloczko/rpmbuild/BUILD/elasticsearch-py-7.14.0/docs/sphinx/helpers.rst:119: WARNING: duplicate object description of elasticsearch.helpers, other instance in async, use :noindex: for one of them
/home/tkloczko/rpmbuild/BUILD/elasticsearch-py-7.14.0/docs/sphinx/index.rst:280: WARNING: Unexpected indentation.
/home/tkloczko/rpmbuild/BUILD/elasticsearch-py-7.14.0/docs/sphinx/index.rst:282: WARNING: Block quote ends without a blank line; unexpected unindent.
/home/tkloczko/rpmbuild/BUILD/elasticsearch-py-7.14.0/docs/sphinx/index.rst:286: WARNING: Unexpected indentation.
/home/tkloczko/rpmbuild/BUILD/elasticsearch-py-7.14.0/docs/sphinx/index.rst:288: WARNING: Block quote ends without a blank line; unexpected unindent.
/home/tkloczko/rpmbuild/BUILD/elasticsearch-py-7.14.0/elasticsearch/connection/http_urllib3.py:docstring of elasticsearch.connection.http_urllib3.Urllib3HttpConnection:1: WARNING: duplicate object description of elasticsearch.connection.http_urllib3.Urllib3HttpConnection, other instance in connection, use :noindex: for one of them
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
writing... python-elasticsearch-py.3 { api exceptions async connection transports helpers } done
build succeeded, 12 warnings.
```

**Comment by sethmlarson** [comment]
We're trying to move away from setuptools being the hub of "all the things" and instead let the individual tools do their own thing. I think in this case it makes sense to remove the `build_sphinx` directive.

The warnings aren't breaking anything, the documentation builds as expected. However if you're interested in cleaning some up that'd be a welcome contribution.

**Comment by kloczek** [comment]
None of the other frameworks takes care of any operations relegated to testing and building documentation.
Things like raw pep517, flit and poetry are actually nothing more than equivalent to `pip wheel` (archivers adding metadata).
You can find my ticket against flit in which I've asked about equivalent of the setuptools<>sphinx integration and maintainer told me that he does not see anything to do on that are. Additionally flit autogenerates from setup.cfg/propject.toml setup.py which is using `distutils` which is already scheduled to be deprecated (at the moment it is used by less than 1% of all modules)
https://www.python.org/dev/peps/pep-0632/#:~:text=In%20Python%203.10%20and%203.11,platforms%20will%20not%20be%20added.

`setuptools` just works but it needs to be correctly done.
In other words please stick to `setuptools` because. The best way is to  keek stub setup.py containing only  'from setuptools import setup; setup()' and moved everything else to serup.cfg/propject.toml (latest setuptools can use both files).
This allows use raw pep517 module to generate .whl on jumpstarting build env and at the same time use setuptools without any problems.
BTW only `setuptools` pep517 backend can handle building for example DSOs (for example over cython).

**Comment by sethmlarson** [comment]
Right but that's not the documented way to build the documentation. You either use `nox -s docs` or do exactly what `nox -s docs` does with `sphinx-build docs/sphinx/ docs/sphinx/_build -b html`. Adding additional ways to build the documentation doesn't make the project any easier to maintain. For another example here we removed a `Makefile` that had been unused in a long time, there's no reason to have additional routes to build the docs.
